### DISCUSSION TITLE
it is possible to give yourself starts... why?

### DISCUSSION THREAD

**Discussion by beta-tester** [community/community#54387] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

what is the point of being able to give yourself stars for your own projects?

**Comment by drefcot17th** [comment]
> ### Select Topic Area
> 
> Question
> 
> ### Body
> 
> what is the point of being able to give yourself stars for your own projects?

**Comment by multikitty** [comment]
Starring a repo means a lot for the repos owner, it will push up the library more, getting more contributors and more issues + feedback. also it encourage the maintainer to spend more time and continue on the project.
The ability to star your own projects on GitHub can be seen as a way to show your support and appreciation for the work you've done. It can also serve as a way to help increase visibility and credibility for your project. Starring a project can make it more discoverable on the platform, and can help to attract other users who are interested in your work. Ultimately, the decision to star your own projects on GitHub is up to the individual user and their goals for their work.

**Comment by multikitty** [comment]
Personally, I've used a self-star as a tiebreaker to call out better projects. If two of my projects both have N stars, and I think one is clearly better than the other, I'll star it so that it ranks over the other one in my profile page. This is especially important if I have many zero-star repositories, because some of them may not appear on my profile page at all (since GitHub limits the list to 5 repos). I might have a zero-star repository that I'd love people to see, but it could be buried in obscurity under a bunch of other zero-star repos.

I personally think it's reasonable to use a self-star to call out what you believe is your best work, especially since the primary place a project's star count is relevant is in ranking your own projects on your profile page.

Does that justification make it currently socially acceptable? I certainly won't judge you; I can't really say whether anybody else will. Anecdotally, though, self-stars appear to be good enough for Linus Torvalds, anyway. (At the time of this writing, he has starred exactly two repos, and they're both his.) Note, however, this may simply be an artifact of when GitHub first introduced stars and automatically converted all "Watch" relationships to stars. See SztupY's answer for more information.

(Ref: https://opensource.stackexchange.com/questions/1918/is-it-socially-acceptable-to-star-your-own-repositories-on-github)

**Comment by beta-tester** [comment]
> ~~Personally, I've used a self-star as a tiebreaker to call out better projects. If two of my projects both have N stars, and I think one is clearly better than the other,~~ I'll star it so that it ranks over the other one in my profile page.

when you are able to add only one '_self-star_' per project, how can you over-rank one project over the other?
e.g.: when one of your projects is having already 200 stars from other peoples and the other only 100 stars.

can you add more than one  '_self-stars_' at a time to increase projects ranking ?!

EDIT: ok, i misread. 200 to 100 is not a tie.

but even then, one single star extra means nothing when your project has already hundreds of stars.

**Comment by woden255** [comment]


**Comment by woden255** [comment]
The others gave great answers. I learned from it.

**Comment by woden255** [comment]
> The others gave great answers. I learned from it.

**Comment by swstl** [comment]
same
